rbd/watchers_mimic.go

```diff
@@ -0,0 +1,98 @@
+// +build !luminous,!mimic
+//
+// Ceph Nautilus is the first version that supports watchers through librbd.
+
+package rbd
+
+// #cgo LDFLAGS: -lrbd
+// #include <errno.h>
+// #include <rbd/librbd.h>
+import "C"
+
+// Handle for a watcher
+// This type was added so that Image.HasWatchers() can be tested.
+type rbdUpdateHandle struct {
```

Comment:
If `rbdUpdateHandle` is private and has exactly one member does it add any value over just internally using some C.uint64_t values directly? If it does, the comment doesn't really explain why.

Reply:
Ah, yes, there were some other attributes while playing with callbacks. The handle is expected to be returned to the consumer of the api, so that it can be used to remove/update the watcher later on. But as all of that is private at the moment, there is no real benefit of the type (anymore).

I'll make `rbdUpdateHandle` a plain `C.uint64_t` so that it can be used in the tests (no CGO allowed).